src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -256,38 +256,35 @@ void activity_classifier::train(gl_sframe data, const std::string& target_column
   float_array_map trained_weights = training_model_->export_weights();
   nn_spec_->update_params(trained_weights);
 
+  variant_map_type state_update;
+
   // Update the state with recall, precision and confusion matrix for training
   // data
   gl_sarray train_predictions = predict(train_data, "probability_vector");
-  variant_map_type train_metric =
-      evaluation::compute_classifier_metrics(
-      train_data, target_column_name, "auto", train_predictions,
+  variant_map_type train_metric = evaluation::compute_classifier_metrics(
+      train_data, target_column_name, "report", train_predictions,
       {{"classes", read_state<flex_list>("classes")}});
 
-  add_or_update_state(
-      {{"training_precision", train_metric["precision"]},
-       {"training_recall", train_metric["recall"]},
-       {"training_accuracy", train_metric["accuracy"]},
-       {"training_log_loss", train_metric["log_loss"]},
-       {"training_confusion_matrix", train_metric["confusion_matrix"]},
-       {"training_number_of_examples", train_data.size()}});
+  for (auto &p : train_metric) {
+    state_update["training_" + p.first] = p.second;
+  }
+  state_update["training_number_of_examples"] = train_data.size();
```

Comment:
Let's skip `number_of_examples` for now, so we're more consistent with the `supervised_learning` toolkit. We could potentially add this metric in both places later